ts/features/messages/components/Home/DS/DoubleAvatar.tsx

```diff
@@ -0,0 +1,168 @@
+import React from "react";
+import {
+  Image,
+  ImageRequireSource,
+  ImageURISource,
+  StyleSheet,
+  View
+} from "react-native";
+import {
+  IOColors,
+  IOIconSizeScale,
+  IOSpacingScale,
+  IOVisualCostants,
+  Icon,
+  hexToRgba,
+  useIOTheme
+} from "@pagopa/io-app-design-system";
+import { addCacheTimestampToUri } from "../../../../../utils/image";
+
+type DoubleAvatarProps = {
+  backgroundLogoUri?:
+    | ImageRequireSource
+    | ImageURISource
+    | ReadonlyArray<ImageURISource>;
+};
+
+const avatarContainerSize: IOIconSizeScale = 30;
+const avatarDoubleRadiusSizeSmall: number = 6;
+const internalSpaceDefaultSize: number = 3;
+const internalSpacePlaceholderDefaultSize: IOSpacingScale = 6;
+const avatarBorderLightMode = hexToRgba(IOColors.black, 0.1);
+
+const styles = StyleSheet.create({
+  topContainer: {
+    height: IOVisualCostants.avatarSizeSmall,
+    width: IOVisualCostants.avatarSizeSmall
+  },
+  avatarWrapper: {
+    overflow: "hidden",
+    borderColor: avatarBorderLightMode,
+    borderWidth: 1,
+    borderCurve: "continuous",
+    position: "absolute"
+  },
+  avatarInnerWrapper: {
+    overflow: "hidden",
+    backgroundColor: IOColors.white,
+    borderCurve: "continuous"
+  },
+  avatarImage: {
+    resizeMode: "contain",
+    height: "100%",
+    width: "100%"
+  }
+});
+
+/**
+ * DoubleAvatar component is used to display the background logo of an organization, with a fixed pagoPA icon on top. It accepts the following props:
+ * - `backgroundLogoUri`: the uri of the image to display. If not provided, a placeholder icon will be displayed. It can be a single uri or an array of uris, in which case the first one that is available will be used.
+ * @param DoubleAvatarProps
+ * @returns
+ */
+export const DoubleAvatar = ({ backgroundLogoUri }: DoubleAvatarProps) => {
+  const theme = useIOTheme();
+  const indexValue = React.useRef<number>(0);
+
+  const [imageSource, setImageSource] = React.useState(
+    backgroundLogoUri === undefined
+      ? undefined
+      : Array.isArray(backgroundLogoUri)
+      ? addCacheTimestampToUri(backgroundLogoUri[0])
+      : typeof backgroundLogoUri === "number"
+      ? backgroundLogoUri
+      : addCacheTimestampToUri(backgroundLogoUri as ImageURISource)
+  );
```

Comment:
```suggestion
  const getImageState = () => {
    switch (typeof backgroundLogoUri) {
      case "number":
        return addCacheTimestampToUri(backgroundLogoUri as ImageURISource);
      case "object":
        if (Array.isArray(backgroundLogoUri)) {
          return addCacheTimestampToUri(backgroundLogoUri[0]);
        } // eslint-disable-next-line no-fallthrough
      default:
        return undefined;
    }
  };
```
  
I believe that concatenated ternary operators get hard to read very fast, so I believe a more verbose approach is better, while maybe less elegant at first sight -- what do you think?

Reply:
Let's also ask to @CrisTofani who did the original implementation

Reply:
LGTM, just be aware hooks should be on top of the component codeso, maybe just define `getImageState` outside of the component

Reply:
Applied